src/layman/user/util.py

```diff
@@ -100,3 +103,116 @@ def get_username_suggestions_from_claims(claims):
 def delete_user(username=None):
     authn_redis.delete_user(username)
     delete_whole_user(username)
+
+
+def check_if_is_only_rw(access_rights, username):
+    read_set = set(access_rights['read'])
+    write_set = set(access_rights['write'])
+    if RIGHTS_EVERYONE_ROLE in read_set or RIGHTS_EVERYONE_ROLE in write_set:
+        return False
+
+    def is_role(item):
+        return item == item.upper()
+    read_users = {u for u in read_set if not is_role(u)}
+    write_users = {u for u in write_set if not is_role(u)}
+    return (read_users == {username}) and (write_users == {username})
+
+
+def delete_user_public_publications(username):
+    def has_only_roles_without_everyone(user_list):
+        return all(u.isupper() and u != RIGHTS_EVERYONE_ROLE for u in user_list)
+
+    skipped_publications = []
+    layers_to_be_deleted = []
+    maps_to_be_deleted = []
+    result = get_publication_infos(
+        context={
+            'actor_name': username,
+            'access_type': 'read',
+        },
+    )
+
+    for (workspace, publication_type, publication_name), pubinfo in result.items():
+        is_public = pubinfo['_is_public_workspace']
+        access_rights = pubinfo['access_rights']
+        geodata_type = pubinfo['geodata_type']
+        is_only_rw = check_if_is_only_rw(access_rights, username)
+        publication_name = pubinfo['name']
+
+        if is_public and is_only_rw:
+            if publication_type == LAYER_TYPE:
+                layers_to_be_deleted.append(((workspace, publication_type, publication_name), pubinfo))
+            elif publication_type == MAP_TYPE:
+                maps_to_be_deleted.append(((workspace, publication_type, publication_name), pubinfo))
```

Comment:
I think you don't need to append `pubinfo`, because it's not needed later. Just `(workspace, publication_type, publication_name)` should be enough.